src/urllib3/util/url.py

```diff
@@ -14,12 +15,12 @@
 NORMALIZABLE_SCHEMES = ('http', 'https', None)
 
 # Regex for detecting URLs with schemes. RFC 3986 Section 3.1
-SCHEME_REGEX = re.compile(r"^[a-zA-Z][a-zA-Z0-9+\-.]*://")
```

Comment:
@sigmavirus24 What do you think of taking this `.` out of this scheme regex? I did this because I don't think we support any scheme that has this `.` here but we support a lot of URLs where the authority section looks like a scheme. (`www.google.com` is a valid "scheme").  Should we get even more strict and only support schemes that start with `http`? I'm not sure.

Reply:
So, I don't understand the purpose the `.` is serving. We can't, however, limit ourselves to what we think of as normal schemes because we (fortunately, or not) support `http+unix://`

Reply:
The `.` is part of the scheme spec but I couldn't find an example scheme that contained a period in it.  The three schemes I know of that we support are `http`, `https`, and `http+unix` case insensitive.